plugins/ecs-bridge/engine/engine.go

```diff
@@ -217,38 +220,54 @@ func (engine *engine) ConfigureContainerVethInterface(netnsName string, result *
 
 // ConfigureBridge configures the IP address of the bridge if needed
 func (engine *engine) ConfigureBridge(result *current.Result, bridge *netlink.Bridge) error {
-	addrs, err := engine.netLink.AddrList(bridge, syscall.AF_INET)
-	if err != nil && err != syscall.ENOENT {
-		return errors.Wrapf(err,
-			"bridge configure: unable to list addresses for bridge %s",
-			bridge.Attrs().Name)
-	}
-
 	resultBridgeNetwork := &net.IPNet{
 		IP:   result.IPs[0].Gateway,
 		Mask: result.IPs[0].Address.Mask,
 	}
 	resultBridgeCIDR := resultBridgeNetwork.String()
-	if len(addrs) > 0 {
-		for _, addr := range addrs {
-			if addr.IPNet.String() == resultBridgeCIDR {
-				return nil
-			}
-		}
-		return errors.New("bridge configure: mismatch in bridge ip address")
+
+	if engine.bridgeNetworkExists(bridge, resultBridgeCIDR) {
+		return nil
 	}
 
 	bridgeAddr := &netlink.Addr{
 		IPNet: resultBridgeNetwork,
 	}
-	err = engine.netLink.AddrAdd(bridge, bridgeAddr)
-	if err != nil {
-		return errors.Wrapf(err,
-			"bridge configure: unable to assign ip address to bridge %s",
-			bridge.Attrs().Name)
+	addrAddErr := engine.netLink.AddrAdd(bridge, bridgeAddr)
+	if addrAddErr == nil {
+		return nil
 	}
 
-	return nil
+	// if we fail to assign ip address, check bridge network exists or not again, because the error may be caused by
+	// someone else assigned the address right before we assigned it
+	if engine.bridgeNetworkExists(bridge, resultBridgeCIDR) {
```

Comment:
This is assuming all the errors are caused by the "bridge already exists" which isn't true. If we ignore the other errors, the bridge may still be created by not working as expected. I think another way is to check the error message only for the conflict. But this should be covered by a integration test in case the error message from your dependency library changed.